packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -118,110 +168,104 @@ contract PluginRepo is
-    /// @return pluginSetup The address of the plugin factory associated with the version.
-    /// @return contentURI The external URI pointing to the content of the version.
-    function getVersionBySemanticVersion(uint16[3] memory _semanticVersion)
+    /// @notice get the version by tag.
+    /// @param _tag the version tag.
+    /// @return the version which belongs to the _tag.
+    function getVersion(Tag calldata _tag)
         public
         view
-        returns (
-            uint16[3] memory semanticVersion,
-            address pluginSetup,
-            bytes memory contentURI
-        )
+        returns (Version memory)
     {
-        return getVersionById(versionIndexForSemantic[semanticVersionHash(_semanticVersion)]);
+        return getVersion(tagHash(_tag));
     }
 
-    /// @notice Gets the version information associated with a version index.
-    /// @return semanticVersion The semantic version number.
-    /// @return pluginSetup The address of the plugin factory associated with the version.
-    /// @return contentURI The external URI pointing to the content of the version.
-    function getVersionById(uint256 _versionIndex)
+    /// @notice get the concrete version.
+    /// @param _tagHash the tag hash.
+    /// @return Version the concrete version by the exact hash.
+    function getVersion(bytes32 _tagHash)
         public
         view
-        returns (
-            uint16[3] memory semanticVersion,
-            address pluginSetup,
-            bytes memory contentURI
-        )
+        returns (Version memory)
     {
-        if (_versionIndex <= 0 || _versionIndex >= nextVersionIndex)
-            revert VersionIndexDoesNotExist({versionIndex: _versionIndex});
-        Version storage version = versions[_versionIndex];
-        return (version.semanticVersion, version.pluginSetup, version.contentURI);
-    }
+        Version memory version = versions[_tagHash];
+
+        if (version.tag.release == 0) {
+            revert VersionHashDoesNotExist(_tagHash);
+        }
 
-    /// @notice Gets the total number of published versions.
-    /// @return uint256 The number of published versions.
-    function getVersionCount() public view returns (uint256) {
-        return nextVersionIndex - 1;
+        return version;
     }
 
-    /// @notice Generates a hash from a semantic version number.
-    /// @param semanticVersion The semantic version number.
-    /// @return bytes32 The hash of the semantic version number.
-    function semanticVersionHash(uint16[3] memory semanticVersion) internal pure returns (bytes32) {
-        return
-            keccak256(abi.encodePacked(semanticVersion[0], semanticVersion[1], semanticVersion[2]));
+    /// @notice Gets the total number of published versions per release.
+    /// @param _release release id.
+    /// @return The number of published versions in the release.
+    function versionCount(uint8 _release) public view returns (uint256) {
```

Comment:
Rename to `buildCount` - otherwise it can be confusing.

Reply:
https://github.com/aragon/core/pull/164/commits/e949b2db3eb214aae5f7ac09ebdfd79980e5aa86

The reason why in mapping I store `uint256` is I also store `uint256 latestRelease`. If I make this uint8, then there will be more conversions. In the end, before merging, I will try to compare costs better and comment here.

Reply:
My second question was wheter the input argument of `function buildCount(uint8 _release)` is `uint8` or should be `uint256`